### PR TITLE
doc: dev model: Add a reference to ncs-example-application

### PR DESCRIPTION
**Pull request by carlescufi**
This new repo is intended to serve as a templated reference of Wokflow 4: application as a manifest repository.

Rework the Workflow 4 section to cover the use of the new ncs-example-application, as well as rewording it for clarity.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

**Comment by carlescufi** [comment]
Note to reviewers: As part of reviewing this PR  please feel free to send PRs modifying the README file in the ncs-example-application repo:
https://github.com/nrfconnect/ncs-example-application#readme

**Review by gmarull — APPROVED** [review]

File: doc/nrf/dm_adding_code.rst [hunk 1]
```diff
@@ -78,8 +80,58 @@ This workflow is particularly beneficial if your application is split among mult
 In order to implement this approach, you first need to create a manifest repository of your own, which just means a repository that contains a :file:`west.yml` manifest file in its root.
 Next you must populate the manifest file with the list of repositories and their revisions.
 
-In general, the easiest thing to do is to import the :file:`west.yml` into `sdk-nrf`_, using west's manifest imports feature.
-This is demonstrated by the following code:
+In order to create your own repository you have a couple of options, described in the sections below.
```
Comment (peknis) [inline doc/nrf/dm_adding_code.rst:83]:
```suggestion
You have a couple of different options to create your own repository.
```

File: doc/nrf/dm_adding_code.rst [hunk 2]
```diff
@@ -78,8 +80,58 @@ This workflow is particularly beneficial if your application is split among mult
...
+
+   west init -m https:\ //github.com/*your-name/your-application* *your-app-worskpace*
+   cd *your-app-workspace*
+   west update
+
+After that, to modify the |NCS| version associated with your app, change the ``revision`` value in the manifest file to the `sdk-nrf`_ Git tag, SHA, or the branch you want to use, save the file, and run ``west update``.
+See :ref:`zephyr:west-basics` for more details.
+
+Recommended: Using the example application repo
+-----------------------------------------------
```
Comment (peknis) [inline doc/nrf/dm_adding_code.rst:99]:
```suggestion
Recommended: Using the example application repository
-----------------------------------------------------
```

File: doc/nrf/dm_adding_code.rst [hunk 3]
```diff
@@ -78,8 +80,58 @@ This workflow is particularly beneficial if your application is split among mult
...
+
+Nordic maintains a templated example repository for the purpose of serving as a reference for users that choose this workflow, `ncs-example-application`_.
```
Comment (peknis) [inline doc/nrf/dm_adding_code.rst:101]:
```suggestion
Nordic Semiconductor maintains a templated example repository as a reference for users that choose this workflow, `ncs-example-application`_.
```

File: doc/nrf/dm_adding_code.rst [hunk 4]
```diff
@@ -78,8 +80,58 @@ This workflow is particularly beneficial if your application is split among mult
...
+This repository showcases many features of both the |NCS| and Zephyr:
```
Comment (peknis) [inline doc/nrf/dm_adding_code.rst:102]:
```suggestion
This repository showcases the following features of both the |NCS| and Zephyr:
```

File: doc/nrf/dm_adding_code.rst [hunk 5]
```diff
@@ -78,8 +80,58 @@ This workflow is particularly beneficial if your application is split among mult
...
+* Out-of-tree libraries
+* Example CI configuration (using Github Actions)
+* Custom :ref:`west extension <zephyr:west-extensions>`
+
+It is highly recommended to use this templated repository as a starting point for
+your own application manifest repo. You can do so by clicking the "Use this
+template" button on the GitHub web user interface.
+Once you have your own copy of the repository you can then modify it at will,
+including adding your own projects to the manifest file, as well as your own boards,
+drivers, and libraries as required by your application.
```
Comment (peknis) [inline doc/nrf/dm_adding_code.rst:119]:
```suggestion
It is highly recommended to use this templated repository as a starting point for your own application manifest repository.
Click the :guilabel:`Use this template` button on the GitHub web user interface.
Once you have your own copy of the repository, you can then modify it.
You can add your own projects to the manifest file, as well as your own boards, drivers, and libraries as required by your application.
```

File: doc/nrf/dm_adding_code.rst [hunk 6]
```diff
@@ -78,8 +80,58 @@ This workflow is particularly beneficial if your application is split among mult
...
+
+The `ncs-example-application`_ repository is tagged every time a new release of
+the |NCS| is made (starting with NCS v2.3.0), using the exact same version
+number. This allows an application developer to select the tag that will match
+the version of the |NCS| that they intend to use.
```
Comment (peknis) [inline doc/nrf/dm_adding_code.rst:124]:
```suggestion
The `ncs-example-application`_ repository is tagged every time a new release of the |NCS| is launched (starting with v2.3.0), using the same version number.
This allows you to select the tag that matches the version of the |NCS| you intend to use.
```

File: doc/nrf/dm_adding_code.rst [hunk 7]
```diff
@@ -78,8 +80,58 @@ This workflow is particularly beneficial if your application is split among mult
...
+
+Creating your own manifest repo from scratch
+--------------------------------------------
```
Comment (peknis) [inline doc/nrf/dm_adding_code.rst:127]:
```suggestion
Creating your own manifest repository from scratch
--------------------------------------------------
```

File: doc/nrf/dm_adding_code.rst [hunk 8]
```diff
@@ -78,8 +80,58 @@ This workflow is particularly beneficial if your application is split among mult
...
+
+If, for whatever reason you decide not to use the example application repository
+as a starting point, then you can start from scratch to create your own manifest repo that imports the |NCS|.
```
Comment (peknis) [inline doc/nrf/dm_adding_code.rst:130]:
```suggestion
If you decide not to use the example application repository as a starting point, you can start from scratch to create your own manifest repository that imports the |NCS|.
```

File: doc/nrf/dm_adding_code.rst [hunk 9]
```diff
@@ -78,8 +80,58 @@ This workflow is particularly beneficial if your application is split among mult
...
+
+In general, the easiest thing to do is to ``import`` `sdk-nrf`_ in your own :file:`west.yml`, using west's :ref:`manifest imports <zephyr:west-manifest-import>` feature.
+This is demonstrated by the following code, that would be placed somewhere in
+your repository (typically in :file:`root/west.yml`):
```
Comment (peknis) [inline doc/nrf/dm_adding_code.rst:134]:
```suggestion
To start the process, ``import`` `sdk-nrf`_ in your own :file:`west.yml`, using the :ref:`manifest imports <zephyr:west-manifest-import>` feature of west.
This is demonstrated by the following code, that would be placed somewhere in your repository (typically in :file:`root/west.yml`):
```

File: doc/nrf/introduction.rst [hunk 10]
```diff
@@ -94,6 +93,20 @@ It also makes it fairly simple to add and remove modules.
 See :ref:`getting_started` for information about how to install the |NCS| and about the first steps.
 See :ref:`dev-model` for more information about the |NCS| code base and how to manage it.
 
+Applications
+************
+
+In order to start developing your application you will need to understand a few
+fundamental concepts first. Follow the :ref:`Zephyr guide to Application Development <zephyr:application>` and browse through the included
+:ref:`reference applications <applications>` in the |NCS| in order to get
+familiar with the basics.
```
Comment (peknis) [inline doc/nrf/introduction.rst:102]:
```suggestion
To start developing your application you need to understand a few fundamental concepts.
Follow the :ref:`Zephyr guide to Application Development <zephyr:application>` and browse through the included :ref:`reference applications <applications>` in the |NCS| to get familiar with the basics.
```

File: doc/nrf/introduction.rst [hunk 11]
```diff
@@ -94,6 +93,20 @@ It also makes it fairly simple to add and remove modules.
...
+
+You will also need to choose how to structure your application. A few
+:ref:`user workflows <dm_user_workflows>` can be chosen from, but having the
+:ref:`application as the manifest repository <dm_workflow_4>` is recommended. To
+that effect an `ncs-example-application`_ repository is provided to serve as a
+reference and/or starting point.
```
Comment (peknis) [inline doc/nrf/introduction.rst:108]:
```suggestion
You also need to decide how to structure your application.
You can choose from a few alternative :ref:`user workflows <dm_user_workflows>`, but having the
:ref:`application as the manifest repository <dm_workflow_4>` is recommended.
An `ncs-example-application`_ repository is provided to serve as a reference or starting point.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**